TS failure: quarkus-integration-test-rest-client-reactive-kotlin-serialization: Unresolved class: interface java.util.List

### Describe the bug

Hello,

The TS needs debug & fix:

```
2022-01-21 14:48:24,689 INFO  [io.quarkus] (main) quarkus-integration-test-rest-client-reactive-kotlin-serialization 2.7.0.CR1 native (powered by Quarkus 2.7.0.CR1) started in 0.028s. Listening on: http://0.0.0.0:8081
2022-01-21 14:48:24,693 INFO  [io.quarkus] (main) Profile prod activated. 
2022-01-21 14:48:24,693 INFO  [io.quarkus] (main) Installed features: [cdi, jaxrs-client-reactive, kotlin, rest-client-reactive, rest-client-reactive-kotlin-serialization, resteasy-reactive, resteasy-reactive-kotlin-serialization, smallrye-context-propagation, vertx]
2022-01-21 14:48:25,974 ERROR [io.qua.ver.htt.run.QuarkusErrorHandler] (vert.x-eventloop-thread-0) HTTP Request to /countries failed, error id: 1aad7d33-167e-47e5-9e37-8355938baf18-1: kotlin.reflect.jvm.internal.KotlinReflectionInternalError: Unresolved class: interface java.util.List
	at kotlin.reflect.jvm.internal.KClassImpl.reportUnresolvedClass(KClassImpl.kt:328)
	at kotlin.reflect.jvm.internal.KClassImpl.access$reportUnresolvedClass(KClassImpl.kt:44)
	at kotlin.reflect.jvm.internal.KClassImpl$Data$descriptor$2.invoke(KClassImpl.kt:56)
	at kotlin.reflect.jvm.internal.KClassImpl$Data$descriptor$2.invoke(KClassImpl.kt:48)
	at kotlin.reflect.jvm.internal.ReflectProperties$LazySoftVal.invoke(ReflectProperties.java:93)
	at kotlin.reflect.jvm.internal.ReflectProperties$Val.getValue(ReflectProperties.java:32)
	at kotlin.reflect.jvm.internal.KClassImpl$Data.getDescriptor(KClassImpl.kt:48)
	at kotlin.reflect.jvm.internal.KClassImpl.getDescriptor(KClassImpl.kt:182)
	at kotlin.reflect.jvm.internal.KClassImpl.getDescriptor(KClassImpl.kt:44)
	at kotlin.reflect.full.KClassifiers.createType(KClassifiers.kt:48)
	at kotlin.reflect.jvm.internal.ReflectionFactoryImpl.typeOf(ReflectionFactoryImpl.java:124)
	at kotlin.jvm.internal.Reflection.typeOf(Reflection.java:132)
	at io.quarkus.it.rest.CountriesResource.init$lambda-3(CountriesResource.kt:80)
	at io.vertx.ext.web.impl.BlockingHandlerDecorator.lambda$handle$0(BlockingHandlerDecorator.java:48)
	at io.vertx.core.impl.ContextImpl.lambda$null$0(ContextImpl.java:159)
	at io.vertx.core.impl.AbstractContext.dispatch(AbstractContext.java:100)
	at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$1(ContextImpl.java:157)
	at io.vertx.core.impl.TaskQueue.run(TaskQueue.java:76)
	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$13.runWith(VertxCoreRecorder.java:543)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:829)
	at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:600)
	at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:192)

2022-01-21 14:48:25,975 ERROR [io.qua.ver.htt.run.QuarkusErrorHandler] (vert.x-eventloop-thread-1) HTTP Request to /call-countries failed, error id: 1aad7d33-167e-47e5-9e37-8355938baf18-2: org.jboss.resteasy.reactive.ClientWebApplicationException: Received: 'Unknown error, status code 500' when invoking: Rest Client method: 'io.quarkus.it.rest.CountriesClient#countries'
	at org.jboss.resteasy.reactive.client.impl.RestClientRequestContext.unwrapException(RestClientRequestContext.java:166)
	at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.handleException(AbstractResteasyReactiveContext.java:313)
	at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:169)
	at org.jboss.resteasy.reactive.client.impl.RestClientRequestContext$1.lambda$execute$0(RestClientRequestContext.java:272)
	at io.vertx.core.impl.AbstractContext.dispatch(AbstractContext.java:100)
	at io.vertx.core.impl.AbstractContext.dispatch(AbstractContext.java:63)
	at io.vertx.core.impl.EventLoopContext.lambda$runOnContext$0(EventLoopContext.java:38)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:469)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:500)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:829)
	at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:600)
	at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:192)
Caused by: javax.ws.rs.WebApplicationException: Unknown error, status code 500
	at io.quarkus.rest.client.reactive.runtime.DefaultMicroprofileRestClientExceptionMapper.toThrowable(DefaultMicroprofileRestClientExceptionMapper.java:17)
	at io.quarkus.rest.client.reactive.runtime.MicroProfileRestClientResponseFilter.filter(MicroProfileRestClientResponseFilter.java:36)
	at org.jboss.resteasy.reactive.client.handlers.ClientResponseFilterRestHandler.handle(ClientResponseFilterRestHandler.java:20)
	at org.jboss.resteasy.reactive.client.handlers.ClientResponseFilterRestHandler.handle(ClientResponseFilterRestHandler.java:9)
	at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:141)
	... 13 more
```

### Expected behavior

Test passes Native or is excluded in Native if it is not meant to pass.

### Actual behavior

_No response_

### How to Reproduce?

```
./mvnw install -Dquickly
./mvnw verify -f integration-tests/pom.xml --fail-at-end --batch-mode -Dno-format -DfailIfNoTests=false -Dnative -pl '!bouncycastle-fips-jsse,!container-image/quarkus-standard-way,!devtools,!google-cloud-functions,!google-cloud-functions-http,!gradle,!kubernetes-client,!kubernetes/maven-invoker-way,!maven,!mongodb-rest-data-panache,!smallrye-opentracing'
```

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

Java Version 11.0.14+9-LTS

### GraalVM version (if different from Java)

native-image 21.3.1.0 Mandrel Distribution

### Quarkus version or git rev

2.7.0.CR1

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/.github/native-tests.json
+++ b/.github/native-tests.json
@@ -99,7 +99,7 @@
         {
             "category": "Misc1",
             "timeout": 65,
-            "test-modules": "maven, jackson, jsonb, kotlin-serialization, quartz, qute, logging-min-level-unset, logging-min-level-set, simple with space",
+            "test-modules": "maven, jackson, jsonb, kotlin-serialization, rest-client-reactive-kotlin-serialization, quartz, qute, logging-min-level-unset, logging-min-level-set, simple with space",
             "os-name": "ubuntu-latest"
         },
         {

--- a/integration-tests/rest-client-reactive-kotlin-serialization/src/main/kotlin/io/quarkus/it/rest/CountriesResource.kt
+++ b/integration-tests/rest-client-reactive-kotlin-serialization/src/main/kotlin/io/quarkus/it/rest/CountriesResource.kt
@@ -10,14 +10,11 @@ import org.eclipse.microprofile.rest.client.RestClientBuilder
 import java.net.URI
 import javax.enterprise.context.ApplicationScoped
 import javax.enterprise.event.Observes
-import javax.inject.Inject
 import javax.ws.rs.Path
 
 @Path("/")
 @ApplicationScoped
-class CountriesResource {
-    @Inject
-    lateinit var json: Json
+class CountriesResource(private val json: Json) {
 
     fun init(@Observes router: Router) {
         router.post().handler(BodyHandler.create())
@@ -50,7 +47,6 @@ class CountriesResource {
             val client = RestClientBuilder.newBuilder()
                 .baseUri(URI.create(rc.body.toString()))
                 .build(CountriesClient::class.java)
-            val result = client.countries()
             rc.response()
                 .setStatusCode(200)
                 .end("OK")
@@ -66,7 +62,7 @@ class CountriesResource {
 
             rc.response()
                 .putHeader("content-type", "application/json")
-                .end(json.encodeToString(body))
+                .end(json.encodeToString("[$body[0]]"))
         }
     }
 }

